warehouse/admin/utils.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from packaging.utils import canonicalize_name
+from pyramid.httpexceptions import HTTPSeeOther
+
+from warehouse.packaging.models import (
+    Project, Release, Dependency, File, Role, JournalEntry, release_classifiers
+)
+
+
+def confirm_project(project, request):
```

Comment:
I think this could be generalized a bit more to cover the other use case at `https://github.com/pypa/warehouse/pull/2539/files#diff-2246831731f72e538a306ad3a32ac674L146` if we set the redirect to `request.current_route_path()` and allowed supplying a custom `message` for the flash at line 26.

Reply:
I was originally hoping to do that, but the blacklist confirmation is slightly different:

* It doesn't require a `project` to exist, it just needs a project name;
* The blacklist form `POST`s to the same route that the form lives at, so it can redirect back to `current_route_path` if the confirmation fails. The delete form, however, just lives on the project detail page, `POST`s to the `admin.project.delete` route, and needs to redirect back to the project detail page instead, since `current_route_path` wouldn't correspond to a page that we could `GET`. (I hope that makes sense)

I left this as a separate utility function so that it could be more easily tested, and potentially reused if we need to confirm other actions for an existing project, but I think it's different enough from the blacklist confirmation that we should let them be separate.